### DISCUSSION TITLE
Default redirect after profil update via email

### DISCUSSION THREAD

**Discussion by ch-lepp** [keycloak/keycloak#16596] [Q&A] [answered]
When a user clicks the "forgot password" link on the login screen, he receives an email.
That email contains a link which he can follow to reset his password.
When the password is reset, the user gets redirected to the client website he originally came from.

When an admin manually triggers the "credential reset", the effected user also receives an email.
That email contains a link which he can follow to reset his password.
When the password is reset, the user does not get redirected, instead a blank screen and the text "Your account has been updated." are shown.

I want the user in that case to be redirected to a default URI.
Is there a way for me to define such a default URI at this point?

Unfortunately, the "Your account has been updated." website is not a fix FTML in the keycloak themes.
The redirect that is triggered in the first case comes from [SessionCodeChecks class](https://github.com/keycloak/keycloak/blob/0f2ca3bfddeeeece06cae4cbd1f7122adb261eed/services/src/main/java/org/keycloak/services/resources/SessionCodeChecks.java#L356).

**Comment by ch-lepp** [comment, marked as answer]
If anyone ever runs into the same problem:

I figured out a workaround.
I changed the themes/my-theme/login/info.ftl file in the following way (new are line 10-12 and line 16-18):

```
<#import "template.ftl" as layout>
<@layout.registrationLayout displayMessage=false; section>
    <#if section = "header">
      <#if messageHeader??>
        ${messageHeader}
      <#else>
        ${message.summary}
      </#if>
    <#elseif section = "form">
      <#if message.summary = msg("accountUpdatedMessage") >
        <#assign overwriteSkipLink = true>
      </#if>
      <div id="kc-info-message">
        <p class="instruction">${message.summary}<#if requiredActions??><#list requiredActions>: <b><#items as reqActionItem>${msg("requiredAction.${reqActionItem}")}<#sep>, </#items></b></#list><#else></#if></p>
        <#if skipLink??>
          <#if overwriteSkipLink>
            <p><a href="https://www.google.de">${kcSanitize(msg("backToApplication"))?no_esc}</a></p>
          </#if>
        <#else>
          <#if pageRedirectUri?has_content && !pageRedirectUri?ends_with("/account/")>
            <p><a href="${pageRedirectUri}">${kcSanitize(msg("backToApplication"))?no_esc}</a></p>
          <#elseif actionUri?has_content>
            <p><a href="${actionUri}">${kcSanitize(msg("proceedWithAction"))?no_esc}</a></p>
          <#elseif (client.baseUrl)?has_content>
            <p><a href="${client.baseUrl}">${kcSanitize(msg("backToApplication"))?no_esc}</a></p>
          </#if>
        </#if>
      </div>
    </#if>
</@layout.registrationLayout>
```

**Comment by mkarenko** [comment]
Dude I frickin love you. Exactly what I was looking for.
